### PR TITLE
[ticket/14950] Add possibility to delete a template block

### PR DESCRIPTION
**Pull request by javiexin**
Adds a new mode to alter_block_array to allow for the deletion of
a certain block of template variables.  The selection method is the
same as for the other modes for alter_block_array.  The passed in
vararray is ignored, and an out of bounds index is considered an error.

PHPBB3-14950

Checklist:

- [ ] Correct branch: master for new features; 3.2.x, 3.1.x for fixes
- [ ] Tests pass
- [ ] Code follows coding guidelines: [master / 3.2.x](https://area51.phpbb.com/docs/master/coding-guidelines.html), [3.1.x](https://area51.phpbb.com/docs/31x/coding-guidelines.html)
- [ ] Commit follows commit message [format](https://wiki.phpbb.com/Git#Commit_Messages)

Tracker ticket:

https://tracker.phpbb.com/browse/PHPBB3-14950

### CODE REVIEW

**Comment by javiexin** [comment]
Added testing, please review.

**Comment by Nicofuma** [comment]
Thanks, I will review it later. Not sure it will make it to 3.2.1 but 3.2.2 surely.

File: tests/template/template_test.php [hunk 1]
```diff
@@ -603,6 +603,53 @@ public function test_more_alter_block_array()
 		$this->assertEquals($expect, str_replace(array("\n", "\r", "\t"), '', $this->display('test')), 'Ensuring S_NUM_ROWS is correct after modification');
 	}
 
+	public function test_delete_alter_block_array()
+	{
+		$this->template->set_filenames(array('test' => 'loop_nested.html'));
+
+		$this->template->assign_var('TEST_MORE', true);
+
+		// @todo Change this
```
Comment (Nicofuma) [inline tests/template/template_test.php]: ?

Comment (javiexin) [inline tests/template/template_test.php]: Copy & paste from https://github.com/javiexin/phpbb/blob/136c74bd1c3d2ef1f22055d9bb14e6f524d41433/tests/template/template_test.php#L558
and https://github.com/javiexin/phpbb/blob/136c74bd1c3d2ef1f22055d9bb14e6f524d41433/tests/template/template_test.php#L581
The reason/explanation might be this one https://github.com/javiexin/phpbb/blob/136c74bd1c3d2ef1f22055d9bb14e6f524d41433/tests/template/template_test.php#L19-L21

**Comment by marc1706** [comment]
Merged via https://github.com/phpbb/phpbb/commit/269150ed40d8cf42c2b8d755191b15229dc170c2

**Pull request closed without merging**